src/custom/pages/Swap/ArrowWrapperLoader.tsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react'
+import styled, { css } from 'styled-components'
+import loadingCowGif from 'assets/cow-swap/cow-load.gif'
+import useLoadingWithTimeout from 'hooks/useLoadingWithTimeout'
+import { useIsQuoteRefreshing } from 'state/price/hooks'
+
+export interface ArrowWrapperProps {
+  children: React.ReactNode
+}
+
+export function ArrowWrapperLoader({ children }: ArrowWrapperProps) {
+  const COW_LOADING_TIME = 4000
+  const isRefreshingQuote = useIsQuoteRefreshing()
+  const showLoader = useLoadingWithTimeout(isRefreshingQuote, COW_LOADING_TIME)
+
+  return (
+    <Wrapper showLoader={showLoader}>
+      {children}
+      <div>{showLoader && <img src={loadingCowGif} alt="Loading prices..." />}</div>
```

Comment:
why keep the div? move the brackets outside the div no? also why do you need the div at all?

Reply:
Actually need it to have a parent for the IMG...apparently Firefox didn't respect the z-index stuff but Chrome did. So the `div` helps to accommodate for that.

Reply:
Leandro mentioned Firefox didn't show the COW gif, so the <div> was a later addition to make it work well/fix on all browsers.

Reply:
hmm adding the div was necessary to get the image z-index to work? interesting

Reply:
I think I initially misread your comment: You probably meant why do we need to include the `div` inside the `showLoader` conditional? We actually only need it to be there for the `<img>` tag, so I moved it inside the brackets. Let me know if that's what you meant.

Otherwise, having the `<div>` itself was required, because of competing z-indexes of the parent's pseudo-elements (::before / ::after on line L96) vs. the `<img>` z-index. There's probably another way, but this seems to work for now...